src/pages/dataTableUtilities/reducer/reducerMethods.js

```diff
@@ -252,3 +262,124 @@ export const sortData = (state, action) => {
   const newData = newSortDataBy(data, newSortBy, columnKeyToDataType[newSortBy], newIsAscending);
   return { ...state, data: newData, sortBy: newSortBy, isAscending: newIsAscending };
 };
+
+/**
+ * Opens a basic modal which does not require any other value
+ * to be set upon opening. i.e. text editors
+ *
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ * Action: { type: 'openBasicModal', modalKey }
+ */
+export const openBasicModal = (state, action) => {
+  const { modalKey } = action;
+  return { ...state, modalIsOpen: true, modalKey };
+};
+
+/**
+ * Sets the modal open state to false, closing any
+ * modal that is open.
+ *
+ * @param {Object} state  The current state
+ * Action: { type: 'closeBasicModal' }
+ */
+export const closeBasicModal = state => ({ ...state, modalIsOpen: false, modalKey: '' });
```

Comment:
Not sure if it's a bad practice, but could use the empty string for modalKey as equivalent of `false` and remove modalIsOpen all together (would have to use a boolean evaluation on the empty string on the modal `open` prop though).

Reply:
well I don't think so, `''` is falsey and it's javascript!

a530db1774b92255e7f02aca0051ada655fbc9fb

Reply:
forgot one a87c3f9d7da9cfea9a6590536c926f58a39f520c